### PR TITLE
Core/Instance: Naxxramas conversation fix

### PR DESCRIPTION
**Pull request by ColdIronWow**
Resolve Issue 1138 - Incorrect conversation between Kel'Thuzad and The Lich King

##### CHANGES PROPOSED:

-  Change the default state of Kel'thuzad's door to 'closed' in SQL
-  Force load Kel'thuzad's room when Sapphiron dies so the RP text begins
-  Fix the RP lines and timing to be the correct for the KT - Lich King conversation
-  Prevent KT from opening the door before the RP is complete

###### ISSUES ADDRESSED:
 Closes https://github.com/azerothcore/azerothcore-wotlk/issues/1138

##### TESTS PERFORMED:
Everything builds.  I tested various scenarios like killing sapphiron and waiting for RP, killing her and running directly to the door before the RP finishes, and ensuring the door works correctly though KT wipes and instance resets.

##### HOW TO TEST THE CHANGES:

- Enter Naxxramas and proceed upwards and kill Sapphiron.
- The RP should begin 10 seconds after Sapphiron dies.
- The door to KT's room should remain closed all the way through the RP, opening when he says "Come heroes"
- The door to KT's room should remain functional through wipes and open after instance resets

##### KNOWN ISSUES AND TODO LIST:
-  The final line ("Come heroes...") has no VO, but it also has no VO on current live. 
-  I'm force-loading KT's grid to get the conversation to start.  This is how Trinity does it, but I feel like there's probably a cleaner way.

##### Target branch(es):

Master

### CODE REVIEW

**Comment by Stoabrogga** [comment]
> I'll suggest just some edits in .sql file ( `` )
> 
> [azerothcore-wotlk/data/sql/updates/pending_db_world/rev_1547686034421572510.sql](https://github.com/azerothcore/azerothcore-wotlk/blob/0b990bfd69ca931ced7c76170cf612678a27a36b/data/sql/updates/pending_db_world/rev_1547686034421572510.sql#L3)
> 
> Line 3 in [0b990bf](/azerothcore/azerothcore-wotlk/commit/0b990bfd69ca931ced7c76170cf612678a27a36b)
>  UPDATE gameobject

And please add a semicolon to the end of the statement.

**Comment by ColdIronWow** [comment]
I added the backticks.  Do you want the UPDATE as a single line as well?

**Comment by poszer** [comment]
> I added the backticks. Do you want the UPDATE as a single line as well?

Yes

**Comment by BarbzYHOOL** [comment]
Awesome @ColdIronWow !!

**Comment by poszer** [comment]
### Tested without this fix:  
When you kill this Boss ( Sapphiron ) the Lich King and Kel'thuzad don't start conversation. 
When you move to the doors where is Kel'thuzad the conversation between LK and KT starting. 
Also, The wrong text appears and sounds too. 
The doors to the Kel'Thuzad are always open which is **totally incorrect**.

### Including this fix:
I can confirm that everything is fine, and everything is **correct**.
When you start to fight Boss  ( Sapphiron ) the doors to the Kel'thuzad are closed ( **Correct** ). 
They are opened when you kill Sapphiron ( **Correct** ). 
The speak between LK and KT starting when you kill Sapphiron ( It's **correct** too ) and when the conversation is over, the another door will be opened too ( **Correct** ).  
And sounds are **correct** too!

I made some images while testing, and i'll include them: 






PS. Nice job @ColdIronWow

File: src/server/scripts/Northrend/Naxxramas/instance_naxxramas.cpp [hunk 1]
```diff
@@ -522,8 +524,11 @@ class instance_naxxramas : public InstanceMapScript
                         heiganAchievement = true;
                     break;
                 case BOSS_SAPPHIRON:
-                    if (state == DONE)
+                    if (state == DONE) {
```
Comment (BarbzYHOOL) [inline src/server/scripts/Northrend/Naxxramas/instance_naxxramas.cpp]:
```suggestion
                    if (state == DONE)
                    {
```

**Comment by BarbzYHOOL** [comment]
thank you @ColdIronWow for the well done PR !

thank you @poszer for quick testing !

**Pull request merged**